src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -3555,6 +3555,11 @@ private static bool IsNegativeConstantForArraySize(BoundExpression expression)
 
         protected BoundExpression BindObjectCreationExpression(ObjectCreationExpressionSyntax node, DiagnosticBag diagnostics)
         {
+            if ((object)node.Type == null)
+            {
+                return new UnboundObjectCreationExpression(node);
```

Comment:
>UnboundObjectCreationExpression [](start = 27, length = 31)

I think we need to bind the arguments here, so that we get errors on them just in case there is no context type to convert them to. The object initializer will unfortunately have to wait, but that occurs less often.

Reply:
My suggestion is to store an `AnalyzedArguments` in the `UnboundObjectCreationExpression`.

---
In reply to: [204929609](https://github.com/dotnet/roslyn/pull/25196#discussion_r204929609) [](ancestors = 204929609)

Reply:
Do we need to Free that `AnalyzedArgumnets`?